### ISSUE TITLE
[BUG] Composition of aspects fails in 0.11.1

### ISSUE THREAD

**Issue by fabiantschirschnitz** [cap-js/cds-typer#103]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Nature of Your Project

TypeScript

### Current Behavior

```
TypeError: Cannot read properties of undefined (reading 'kind')
    at Visitor.visitEntity (C:\Users\Tschirschnitz\git\issue-cds-typer\node_modules\@cap-js\cds-typer\lib\visitor.js:389:24)
    at Visitor.visitDefinitions (C:\Users\Tschirschnitz\git\issue-cds-typer\node_modules\@cap-js\cds-typer\lib\visitor.js:119:22)
    at new Visitor (C:\Users\Tschirschnitz\git\issue-cds-typer\node_modules\@cap-js\cds-typer\lib\visitor.js:83:14)
    at compileFromCSN (C:\Users\Tschirschnitz\git\issue-cds-typer\node_modules\@cap-js\cds-typer\lib\compile.js:68:23)
    at compileFromFile (C:\Users\Tschirschnitz\git\issue-cds-typer\node_modules\@cap-js\cds-typer\lib\compile.js:50:12)
```

### Expected Behavior

Types are generated.

### Steps To Reproduce

```cds
entity Foo : cuid {
    @mandatory
    name : String(80);
    bars : Composition of many Bar;
}

aspect Bar : cuid {
    name : String(80);
}
```

if you change Bar to `entity` everything works fine. this was already working in cds-typer 0.10 btw.

### Environment

```markdown
- **OS**: WIN 11
- **Node**: v18.16.1
- **npm**: 9.5.1
- **cds-typer**: 0.11.1
- **cds**: 7.2.1
```

### Repository Containing a Minimal Reproducible Example

https://github.com/fabiantschirschnitz/issue_cds_typer/tree/aspect_error

**Comment by fabiantschirschnitz** [comment]
fyi @nlaenger

**Comment by daogrady** [comment]
Hi Fabian,

thanks for reporting this issue. I have created [a PR](https://github.com/cap-js/cds-typer/pull/106) that addresses the problem. Feel free to check if it fixes the issue in your actual model before I merge it.

The issue stems from a design flaw that is currently plaguing cds-typer where we have to carry two CSN flavours for various lookups. I have added a bandaid fix for now, but I fully expect the issue to pop up again once we tackle said design flaw. So feel free to report back once this resurfaces!

Best,
Daniel
